OpenRA.Mods.Common/UpdateRules/Rules/20190314/RefactorHarvesterIdle.cs

```diff
@@ -0,0 +1,49 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class RefactorHarvesterIdle : UpdateRule
+	{
+		public override string Name { get { return "Refactor harvester idle behavior."; } }
+		public override string Description
+		{
+			get
+			{
+				return "The MaxIdleDuration parameter has been removed from the Harvester trait as part of a\n" +
+					   " refactoring of harvester idling behavior.";
+			}
+		}
+
+		readonly List<string> locations = new List<string>();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (locations.Any())
+				yield return "The MaxIdleDuration parameter has been removed from the harvester logic on the following actors:\n" +
```

Comment:
We typically only explicitly report locations where we expect modders to have to make manual changes. I guess there is no harm here, but we could simplify the rule by silently removing the node instead.

Reply:
Removed.